src/formatDate.js

```diff
@@ -8,7 +8,35 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  // eslint-disable-next-line max-len
+  const dateParts = date.split(fromFormat[3]); // Використовуємо роздільник з fromFormat
+
+  // Створюємо об'єкт для зручного доступу до частин дати
+  const dateObj = {};
+
+  fromFormat.slice(0, 3).forEach((format, index) => {
+    dateObj[format] = dateParts[index];
+  });
+
+  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    if (dateObj['YY']) {
+      const year = parseInt(dateObj['YY'], 10);
+
+      dateObj['YYYY'] = year < 30 ? `20${dateObj['YY']}` : `19${dateObj['YY']}`;
+    }
```

Comment:
The check for `dateObj['YY']` is redundant here since you've already assigned the parts to `dateObj` and checked the formats. You can safely assume that if `fromFormat` includes 'YY', then `dateObj['YY']` exists.